src/Versity.Products/Core/Application/RequestHandlers/Queries/GetAllProducts/GetAllProductsRequestHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Application.Abstractions.Repositories;
+using Domain.Models;
+using FluentResults;
+using MediatR;
+
+namespace Application.RequestHandlers.Queries.GetAllProducts;
+
+public class GetAllProductsRequestHandler : IRequestHandler<GetAllProductsQuery, Result<IEnumerable<Product>>>
+{
+    private readonly IVersityProductsRepository _products;
+
+    public GetAllProductsRequestHandler(IVersityProductsRepository products)
+    {
+        _products = products;
+    }
+
+    public Task<Result<IEnumerable<Product>>> Handle(GetAllProductsQuery request, CancellationToken cancellationToken)
+    {
+        var products = _products
+            .GetAllProducts()
+            .OrderBy(x => x.Title)
+            .Skip(10 * (request.Page - 1))
```

Comment:
I think i saw it

Reply:
the same way i do [in this command handler](https://github.com/jejikeh/versity/pull/17/files#diff-4b98c05264f69a3e6d1e1c8850465d3d8c5993b2f4e85b65d42c497b6099cbb5L7)